### PR TITLE
Ensure the right committee (un)supports a citizen project

### PR DESCRIPTION
**Pull request by thomasbisignani**
**TODO** : Update and add some tests to coverage added and deleted supports.

### CODE REVIEW

File: src/Repository/CitizenProjectCommitteeSupportRepository.php [hunk 1]
```diff
@@ -2,14 +2,15 @@
...
 
+use AppBundle\Entity\CitizenProject;
 use AppBundle\Entity\CitizenProjectCommitteeSupport;
 use AppBundle\Entity\Committee;
 use Doctrine\ORM\EntityRepository;
 
 class CitizenProjectCommitteeSupportRepository extends EntityRepository
 {
-    public function findByCommittee(Committee $committee): ?CitizenProjectCommitteeSupport
+    public function findByCommittee(Committee $committee, CitizenProject $citizenProject): ?CitizenProjectCommitteeSupport
```
Comment (Remg) [inline src/Repository/CitizenProjectCommitteeSupportRepository.php]: you should adjust this method name now

File: src/Repository/CitizenProjectCommitteeSupportRepository.php [hunk 2]
```diff
@@ -2,14 +2,15 @@
...
 
+use AppBundle\Entity\CitizenProject;
 use AppBundle\Entity\CitizenProjectCommitteeSupport;
 use AppBundle\Entity\Committee;
 use Doctrine\ORM\EntityRepository;
 
 class CitizenProjectCommitteeSupportRepository extends EntityRepository
 {
-    public function findByCommittee(Committee $committee): ?CitizenProjectCommitteeSupport
+    public function findByCommitteeAndCitizenProject(Committee $committee, CitizenProject $citizenProject): ?CitizenProjectCommitteeSupport
```
Comment (Remg) [inline src/Repository/CitizenProjectCommitteeSupportRepository.php]: this should be `findOneByXXX` if you return a single entity

Comment (Remg) [inline src/Repository/CitizenProjectCommitteeSupportRepository.php]: this should be `findOneByXXX` if you return a single entity

**Comment by Remg** [comment]
i will add tests 👌

File: src/Controller/EnMarche/CitizenProjectController.php [hunk 3]
```diff
@@ -177,16 +177,21 @@ public function committeeSupportAction(Request $request, CitizenProject $citizen
...
+                $flashMessage = sprintf(
+                    'Votre comité %s soutient maintenant le projet citoyen %s',
+                    $committee->getName(),
+                    $citizenProject->getName()
+                );
             } catch (CitizenProjectCommitteeSupportAlreadySupportException $committeeSupportAlreadySupportException) {
-                $citizenProjectManager->deleteCommitteeSupport($committee);
+                $citizenProjectManager->deleteCommitteeSupport($committee, $citizenProject);
                 $flashMessage = sprintf(
                     'Votre comité %s ne soutient plus le projet citoyen %s',
```
Comment (B-Galati) [inline src/Controller/EnMarche/CitizenProjectController.php]: Wrong error message no ?

Comment (Remg) [reply src/Controller/EnMarche/CitizenProjectController.php]: i don't think so ?

Comment (B-Galati) [reply src/Controller/EnMarche/CitizenProjectController.php]: Got it thanks ;-)

**Pull request merged**